test/handlescope.cc

```diff
@@ -13,6 +13,16 @@ Value createScope(const CallbackInfo& info) {
   return String::New(info.Env(), "scope");
 }
 
+Value createScopeFromExisting(const CallbackInfo& info) {
+  {
+    napi_handle_scope scope;
+    napi_open_handle_scope(info.Env(), &scope);
```

Comment:
Could you have used a second HandleScope declaration here instead of making the node-api call directly?  I think it would then look like

HandleScope scope_one(info.Env())
HandleScope scope_two(info.Env, scope_one.napi_handle_scope())

Reply:
I think I've tried that approach before, but ran into some issues with `~HandleScope` being called on the same underly `napi_handle_scope` twice when they go out of scope.

Reply:
@JckXia if that's the case then I think we might have a real problem?

Reply:
Hmm, I believe the `HandleScope`constructor where we provide an existing `napi_handle_scope` simply saves a reference of the `napi_handle_scope` passed in, as opposed to something like a move constructor where we set the `other` to null. 

I think this means when `scope_two` goes out of scope, its dtor gets ran,  we successfully close the referenced handle scope. But when `scope_one` goes out of scope, its dtor will attempt to call `napi_close_handle_scope` on what I assume is an already closed handle scope, resulting in a `napi_handle_scope_mismatch` error. 

The reason I created the `napi_handle_scope` via a direct call to node-api is such that only scope_two's dtor will be ran, closing it successfully.